tools/ci/push_docs_to_repo.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import argparse
+import subprocess
+import tempfile
+import os
+import sys
+import shutil
+
+
+parser = argparse.ArgumentParser(
+    description='Upload files to a remote repo, replacing existing content'
+)
+parser.add_argument('dir', help='directory of which content will be uploaded')
+parser.add_argument('remote', help='remote to which content will be pushed')
+parser.add_argument('--message', default='Commit bot upload',
+                    help='commit message to use')
+parser.add_argument('--committer', default='numpy-commit-bot',
+                    help='Name of the git committer')
+parser.add_argument('--email', default='numpy-commit-bot@nomail',
+                    help='Email of the git committer')
+
+parser.add_argument(
+    '--force', action='store_true',
+    help='hereby acknowledge that remote repo content will be overwritten'
+)
+args = parser.parse_args()
+args.dir = os.path.abspath(args.dir)
+
+if not os.path.exists(args.dir):
+    print('Content directory does not exist')
+    sys.exit(1)
+
+
+def run(cmd, stdout=True):
+    pipe = None if stdout else subprocess.DEVNULL
+    try:
+        subprocess.check_call(cmd, stdout=pipe, stderr=pipe)
+    except subprocess.CalledProcessError:
+        print("\n! Error executing: `%s;` aborting" % ' '.join(cmd))
+        sys.exit(1)
+
+
+workdir = tempfile.mkdtemp()
+os.chdir(workdir)
+
+run(['git', 'init'])
```

Comment:
Is there a reason we create a new history every time?

Reply:
It's easier to start over than to correctly make the update in place (you have to worry about things like deleting files that shouldn't exist anymore, etc.). And there's no point in wasting space and time fetching and storing old versions of the rendered HTML; we already have the .rst file history.